### PR TITLE
Add custom error decoder example, use apache http client for feign

### PR DESCRIPTION
**Pull request by artemsnisarenko**
https://github.com/spring-cloud/spring-cloud-netflix/issues/1811
https://bugs.openjdk.java.net/browse/JDK-8052118

### CODE REVIEW

File: xs2a-adapter-service-remote/src/test/java/de/adorsys/xs2a/tpp/ErrorResponseDecoder.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    @Override
+    public Exception decode(String methodKey, Response response) {
+        ResponseHeaders responseHeaders = ResponseHeaders.fromMap(
+            response.headers().entrySet().stream()
+                .collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue().iterator().next()))
+        );
+
+        ErrorResponse errorResponse = null;
+        try {
+            errorResponse = objectMapper.readValue(response.body().toString(), ErrorResponse.class);
```
Comment (tadschik) [inline xs2a-adapter-service-remote/src/test/java/de/adorsys/xs2a/tpp/ErrorResponseDecoder.java]: response.body().toString() is throwing JsonParseException
should be response.body().asInputStream()

Comment (artemsnisarenko) [reply xs2a-adapter-service-remote/src/test/java/de/adorsys/xs2a/tpp/ErrorResponseDecoder.java]: Could you provide a reproducible scenario?

Comment (tadschik) [reply xs2a-adapter-service-remote/src/test/java/de/adorsys/xs2a/tpp/ErrorResponseDecoder.java]: response.body() return value is of type (interface) feign.Body. This class defines methods asInputStream() and asReader() for extracting the response. The implemantation class feign.InputStreamBody for example does not override the toString() method.

See also
https://www.codota.com/code/java/methods/feign.Response/body

Comment (artemsnisarenko) [reply xs2a-adapter-service-remote/src/test/java/de/adorsys/xs2a/tpp/ErrorResponseDecoder.java]: I see your point.
https://github.com/adorsys/xs2a-adapter/pull/177

**Pull request merged**